Models/library.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative 'author'
+require_relative 'book'
+require_relative 'order'
+require_relative 'reader'
+require 'yaml'
+
+class Library
+  attr_reader :books, :orders, :authors, :readers
+
+  def initialize(authors, books, readers, orders)
+    @authors = authors
+    @books = books
+    @readers = readers
+    @orders = orders
+  end
+
+  def read_from_yaml(file_name = 'library.yml')
+    if File.exist?(file_name)
+      library = YAML.load_file(file_name)
+      @authors = library.authors
+      @books = library.books
+      @orders = library.orders
+      @readers = library.readers
+    else
+      puts 'There is no such file.'
+    end
+  end
+
+  def write_to_yaml(file_name = 'library.yml')
+    File.delete(file_name) if File.exist?(file_name)
+    File.new(file_name, 'w')
+    File.open(file_name, 'w') { |file| file.write(to_yaml) }
```

Comment:
It is not nessesary to delete and create NEW file.
You can use  File.write(file_name, self.to_yaml),  it create file if file doen`t exist and rewrite all data inside if it already present

Reply:
Thanks a lot!! Fixed